zerver/views/integrations.py

```diff
@@ -131,7 +131,10 @@ def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
 
 
 @has_request_variables
-def integration_doc(request: HttpRequest, integration_name: str=REQ(default=None)) -> HttpResponse:
+def integration_doc(request: HttpRequest,
+                    integration_name: Optional[str]=REQ(default=None, type=str)) -> HttpResponse:
```

Comment:
Yeah, I'm pretty sure that from the pattern in `zproject/urls.py` that points here, this has to be non-None.

Reply:
OK, I've adjusted this. For future reference, is it obvious what can lead to `None`?

Reply:
Sure. This URL route has a group for `integration_name` right in the regex for the route:
```
    url(r'^integrations/doc-html/(?P<integration_name>[^/]*)$',
        zerver.views.integrations.integration_doc,
        name="zerver.views.integrations.integration_doc"),
```
And that group has to be there for the pattern to match -- it's not followed by a `?`, or on one side of a `|`, or affected by any other regex operator that could let the overall pattern match without having that bit match something.

So, then, whatever matches that group gets supplied as the argument `integration_name`.

There are some other routes where the group is optional:
```
    url(r'^accounts/send_confirm/(?P<email>[\S]+)?',
        TemplateView.as_view(template_name='zerver/accounts_send_confirm.html'),
        name='send_confirm'),
```
and we also often have routes with arguments that aren't in the URL path at all, but in say the POST body. Then those can be missing and the URL router will still call the underlying function to handle it, passing None.